server/controllers/documentController.js

```diff
@@ -1,132 +1,298 @@
+          });
+        })
+        .catch((error) => {
+          response.status(401).send({
+            success: false,
+            message: error.message
+          });
+        });
+    } else {
+      response.status(404).send({
+        success: false,
+        message: 'Some fields are missing'
+      });
     }
-    /**
-     * Method used to create new Document for a particular user
-     * @param{Object} app - Express app
-     * @returns{response data}
-     */
-    static createDocument(request, response) {
-        if (DocumentController.postRequest(request)) {
-            return Documents
-                .create({
-                    title: request.body.title,
-                    content: request.body.content,
-                    access: request.body.access,
-                    OwnerId: request.body.OwnerId,
-                })
-                .then(document => response.status(201).send(document))
-                .catch(error => response.status(401).send(error));
+  }
+  /**
+   * Fetch specific document in the database
+   * Admin has access to all the documents
+   * Users only have access to their private documents and all other public documents.
+   * @param{Object} request - Server request
+   * @param{Object} response - Server response
+   * @return {Void} - returns Void
+   */
+  static fetchDocuments(request, response) {
+    Documents.findOne({
+      where: {
+        id: request.params.id
+      },
+      include: [{
+        model: db.users,
+        attributes: ['RoleId']
+      }]
+    }).then((result) => {
+      const document = result ? result.dataValues : null;
+      if (document) {
+        // if the requester's role id is Admin, allow unrestricted access
+        if (request.decoded.RoleId === 1) {
+          response.status(200).send({
+            success: true,
+            message: 'Document found',
+            document
+          });
+        } else if ((document.access === 'public' ||
+          document.user.dataValues.RoleId === request.decoded.RoleId)
+          && document.access !== 'private') {
+          response.status(200).send({
+            success: true,
+            message: 'Document Found',
+            document
+          });
+        } else if (document.access === 'private' && document.OwnerId === request.decoded.UserId) {
+          response.status(200).send({
+            success: true,
+            message: 'Document Found',
+            document
+          });
         } else {
-            response.status(404).send({
-                success: false,
-                message: 'some fields are missing'
-            });
+          response.status(403).send({
+            success: false,
+            message: 'Forbidden!, You are forbidden to access this document'
+          });
         }
+      } else {
+        response.status(404).send({
+          success: false,
+          message: `Document with id ${request.params.id} not found in the database`
```

Comment:
You dont have to return the `id`. A `document not found` message is good enough.